### PR TITLE
Allow variable loan amounts; closes #174

### LINKED ISSUE #174: A project has multiple loans, each of which has an amount greater >= $10 and <= remaining-funding-needed

### PR DESCRIPTION
**Pull request by EmilyMB**
This is the first step to allow lenders to make loans with varying amount. This solves the database issues. The next step will be to address the users stories associated with this change #163.

NOTE: must rake db:drop db:create db:migrate db:seed

Changes include:
* Add/modify validations to ensure that loan amounts could not be less than $10 or greater than the amount still needed for a given project.
* Add "amount" column to loans table
* Add "amount_needed" column to projects table

### CODE REVIEW

File: test/models/loan_test.rb [hunk 1]
```diff
@@ -10,13 +10,42 @@ class LoanTest < ActiveSupport::TestCase
...
+
+    assert loan.invalid?
+  end
+
+  test "it is not valid without an amount" do
+    loan = create(:loan)
+    loan.update_attributes(amount: nil)
+
+    assert loan.invalid?
+  end
```
Comment (mikedao) [inline test/models/loan_test.rb]: I thought previously there was a validation that set loan amount to a certain default?

Comment (EmilyMB) [inline test/models/loan_test.rb]: Previously the loan didn't have an amount.

File: db/seeds.rb [hunk 2]
```diff
@@ -201,31 +201,13 @@
...
-                  total_cost: project4.price,
-                  user_id:    lender.id,
-                  status:     "ordered"
-                  )
+4.times do |x|
+  project = Project.find(x + 1)
+  order =  Order.create!(
+                        user_id: lender.id,
+                        status:  "ordered",
+                        loans:   []
```
Comment (EmilyMB) [inline db/seeds.rb]: I am leaving the spacing so the attributes are aligned.

File: app/models/loan.rb [hunk 3]
```diff
@@ -3,4 +3,19 @@ class Loan < ActiveRecord::Base
...
+    {
+      greater_than_or_equal_to: 1000,
+      less_than_or_equal_to: :project_amount_needed
+    }
+
+  private
+
+  def project_amount_needed
+    if project.nil?
+      1000
```
Comment (mikedao) [inline app/models/loan.rb]: I don't understand this, if there is no project, the amount needed is 10 dollars?

Comment (EmilyMB) [inline app/models/loan.rb]: Because of the way the validation works, I needed the less_than_or_equal_to to be larger than the greater_than_or_equal_to.

Comment (EmilyMB) [inline app/models/loan.rb]: I welcome suggestions on how to fix this, but with the tests in test/models/loan_test.rb, I tried many different combinations, and this was the only way I figured out to test attributes with the validations in place.

**Comment by AdamSmith910** [comment]
Tests pass and look good.  Any questions I had were covered by Mike already.  Good job

**Pull request merged**